### PR TITLE
feat: tabs component

### LINKED ISSUE #384: Page tabs for asset detail view
- Use https://carbon-react-next.netlify.app/?path=/story/components-tabs--default
- Stub placeholder sub-pages for now (Usage, Design, Code)

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #384 

- this PR adds the tabs to asset detail page. 

## Sketch file redlines: 

- [Box file: Component Asset Detail Page Redlines](https://ibm.ent.box.com/folder/150822549524)

## Note:

- The tabs for v11 do not receive hover state at the moment (known. issue) current PR does not contain hover styles

### CODE REVIEW

**Comment by alisonjoseph** [comment]
This is a design question, why are we not using Carbon tabs out of the box?

Quick local test of what it looks like with default Carbon styles
Screen Shot 2022-03-16 at 9 50 01 PM

File: services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].module.scss [hunk 1]
```diff
@@ -35,3 +35,37 @@
...
+    border-bottom: $spacing-02 solid var(--tab-border-color, transparent);
+    color: $text-primary;
+    @include type-style('body-short-02', true);
+  }
+
+  :global(.cds--tabs__nav-item--selected) {
+    border-bottom: $spacing-02 solid $border-interactive;
+    font-weight: 600;
+  }
+}
```
Comment (alisonjoseph) [inline services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].module.scss:71]: I'd recommend removing all of this and using core Carbon styles. (need to sync w/ design)

File: services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].module.scss [hunk 2]
```diff
@@ -35,3 +35,37 @@
...
+    margin-left: -$spacing-05;
+  }
+
+  &::before {
+    position: absolute;
+    z-index: 0;
+    left: 0;
+    width: 100vw;
+    height: $spacing-09;
+    border-bottom: 1px solid $layer-active-01;
```
Comment (alisonjoseph) [inline services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].module.scss:50]: Would we want to just use the pixel height here?. For the border-bottom, is there a border token that would make more sense?

**Review by francinelucca — DISMISSED** [review]
Looks visually correct to me.
Content block horizontal

Just comment on repeated tabs, if design says we're good with carbon overrides then LGTM :tada

**Review by mattrosno — COMMENTED** [review]
- [ ] This right chevron isn't clickable?

image

- [x] Should the tab bar be sticky on scroll (immediately under global nav)?
- [x] Are there supposed to be hover states?
- [x] Can we put these in a new `PageTabs` component that passes in an array of tabs?

Something like this makes sense to me:

```jsx
PageTabs.propTypes = {
  className: PropTypes.string,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      path: PropTypes.string.isRequired
    })
  )
}
```

**Review by mattrosno — COMMENTED** [review]
Should page tabs stack like this on scroll? @kingtraceyj 

image

**Comment by kingtraceyj** [comment]
yes, those tabs are sticky on scroll like they are today on the site.

File: services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js [hunk 3]
```diff
@@ -87,6 +87,20 @@ const Asset = ({ libraryData }) => {
...
+      <Grid className={styles.tabsContainer}>
+        <Column sm={4} md={8}>
+          <Tabs>
+            <TabList aria-label="List of tabs" className={styles.tabList}>
+              <Tab>Usage</Tab>
+              <Tab>Design</Tab>
+              <Tab>Code</Tab>
+              <Tab>Usage</Tab>
+              <Tab>Design</Tab>
+              <Tab>Code</Tab>
```
[the remark this answers is not in the document]
Comment (andreancardona) [reply services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js:99]: si. :)

**Comment by andreancardona** [comment]
> * an we put these in

- I have a note in the PR description but Tabs in v11 have a hover bug (known). Will inquire more. :) 
- Not sure what's going on with the click scroll in mobile - will also inquire

- [x] Move tabs to `PageTabs` component - sorry about this should have done this from the start

File: services/web-app/components/page-tabs/page-tabs.module.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    border-bottom: $spacing-02 solid var(--tab-border-color, transparent);
+    color: $text-secondary;
+    @include type-style('body-short-02', true);
+
+    &:hover {
+      border-bottom: $spacing-02 solid $layer-active-01;
+      color: $text-primary;
+    }
+  }
+}
```
Comment (alisonjoseph) [inline services/web-app/components/page-tabs/page-tabs.module.scss:45]:
```suggestion
:global(.cds--tabs .cds--tab--list) {
  width: auto;
}
```

File: services/web-app/components/page-tabs/page-tabs.module.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  @include breakpoint(md) {
+    margin-left: -$spacing-05;
+  }
+
+  &::before {
+    position: absolute;
+    z-index: 0;
+    left: 0;
+    width: 100vw;
+    height: $spacing-09;
```
Comment (alisonjoseph) [inline services/web-app/components/page-tabs/page-tabs.module.scss:17]:
```suggestion
    height: $spacing-08;
```

to match default tab height

**Review by francinelucca — CHANGES_REQUESTED** [review]

File: services/web-app/components/page-tabs/page-tabs.module.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
+//
+// Copyright IBM Corp. 2022, 2022
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+.container {
+  @include breakpoint(md) {
+    margin-left: -$spacing-05;
```
[the remark this answers is not in the document]
Comment (andreancardona) [reply services/web-app/components/page-tabs/page-tabs.module.scss]: yess thank you, changes made!

**Comment by andreancardona** [comment]
@kingtraceyj 

This is where the tabs preview is at now. There are 2 known bugs with Carbon v11 Tabs. 
1. Text is _not_ left-aligned (currently centered)

2. On focus text jumps up by 1px because padding is not embedded. 

Let me know if there is anything else you see that needs to be updated

https://user-images.githubusercontent.com/32720851/159508353-be1e252c-188d-4c3a-8fe0-c51e7a993379.mov

File: services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js [hunk 7]
```diff
@@ -111,11 +115,13 @@ const Asset = ({ libraryData }) => {
                   <dt className={dashboardStyles.label}>Library</dt>
                   <dd className={dashboardStyles.labelLarge}>{libraryData.content.name}</dd>
                 </dl>
-                <Link href={libraryPath}>
-                  <a className={clsx(dashboardStyles.metaLink, dashboardStyles.metaLinkLarge)}>
-                    {`v${libraryData.content.version}`}
-                  </a>
-                </Link>
+                <CarbonLink
```
Comment (alisonjoseph) [inline services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js:118]: Tabs look great! I think we need to wrap the CarbonLink in the Next js Link component, is that correct @mattrosno 

I actually just opened a small PR here to fix the styles https://github.com/carbon-design-system/carbon-platform/pull/407

Comment (andreancardona) [reply services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js:118]: and so in this case - yep they would need to be wrapped. I'll remove my update from this PR so it doesn't conflict with yours, which is the correct solution.  :)

**Review by alisonjoseph — DISMISSED** [review]
🥳 LGTM!

**Comment by andreancardona** [comment]
Things to note: 

- Currently the scroll click in mobile does not work - the bug is known and the issue is reported: https://github.com/carbon-design-system/carbon/issues/11037

File: services/web-app/components/page-tabs/page-tabs.module.scss [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    @include type.type-style('body-short-02', true);
+
+    &:hover:not(:focus) {
+      border-bottom: spacing.$spacing-02 solid theme.$layer-active-01;
+      color: theme.$text-primary;
+    }
+  }
+
+  /* stylelint-disable-next-line no-descending-specificity */
+  :global(.cds--tabs__nav-item--selected) {
```
Comment (francinelucca) [inline services/web-app/components/page-tabs/page-tabs.module.scss:47]: is this a carbon override? should we add an override comment like in https://github.com/carbon-design-system/carbon-platform/blob/main/services/web-app/components/catalog-sort/catalog-sort.module.scss#L79

**Review by francinelucca — DISMISSED** [review]
🎉

**Comment by mattrosno** [comment]
@andreancardona can you request new reviews after you merge in main branch?

**Comment by andreancardona** [comment]
> I'm seeing this locally now after pulling in main.
> 
> Screen Shot 2022-03-24 at 12 28 37 PM

hmm looking into this now

**Comment by andreancardona** [comment]
@alisonjoseph fixed - ready for your review!

**Review by alisonjoseph — DISMISSED** [review]
LGTM!

File: services/web-app/components/page-tabs/page-tabs.module.scss [hunk 9]
```diff
@@ -0,0 +1,49 @@
+//
+// Copyright IBM Corp. 2022, 2022
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
```
Comment (jdharvey-ibm) [inline services/web-app/components/page-tabs/page-tabs.module.scss:6]:
```suggestion
/*
 * Copyright IBM Corp. 2022, 2022
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
```

File: services/web-app/components/page-tabs/page-tabs.module.scss [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    }
+  }
+
+  /* stylelint-disable-next-line no-descending-specificity */
+  :global(.cds--tabs__nav-item--selected) {
+    border-bottom: spacing.$spacing-02 solid theme.$border-interactive;
+    @include type.type-style('heading-02');
+
+    color: theme.$text-primary;
+  }
```
Comment (jdharvey-ibm) [inline services/web-app/components/page-tabs/page-tabs.module.scss:48]: Could you add a comment for why this override is needed? I read through the stylelint docs for it and it looks like maybe this is a limitation of stylelint? It seems to not be recognizing that the two global blocks are actually targeting different class names.

**Review by alisonjoseph — APPROVED** [review]

**Review by jharvey10 — APPROVED** [review]

**Pull request merged**